page-objects/pages/TASKListPage.js

```diff
@@ -160,67 +147,71 @@ async function clickDeleteTaskButtonInPopup() {
-	await waitHelpers.waitUntilElementIsVisible(taskOverviewLoad);
-	await driver.pause(3000);
-	const taskOverviewResult = await getNuxtTaskList();
-	const taskIndex = taskOverviewResult.indexOf(taskName);
-	let clickOnTask = taskOverviewResult[taskIndex];
-	const taskInTheList = '//div[text() =' + "'" + clickOnTask + "'" + ']';
-	return taskInTheList;
+async function getTaskFromTaskOverview(taskName) {
+	let tasksOnThePage = [];
+	await waitHelpers.waitUntilNuxtClientLoads();
+	await waitHelpers.waitUntilElementIsPresent(taskSection);
+	await driver.$(taskSection).$$(taskTitle).forEach(async function (element)  {
+		tasksOnThePage.push(await element.getText());
+	})
+	let isTaskInTheList = new Boolean(false);
+	isTaskInTheList =  (tasksOnThePage.includes(taskName)) ? driver.$(mod_extsprintf.sprintf(taskTitleText, taskName)) : false ;
+	return isTaskInTheList
 }
 
 async function taskDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
 	await waitHelpers.waitUntilElementIsPresent(taskInTheList);
 }
 
 async function taskNotDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
-	await waitHelpers.waitUntilElementIsNotPresent(taskInTheList);
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
+	expect(taskInTheList).to.equal(false);
 }
 
 async function studentSubmittedTask() {
 	await elementHelpers.clickAndWait(submittedTask);
 	await elementHelpers.clickAndWait(studentSubmitTask);
 }
 
-async function getNuxtTaskList() {
-	const listOfAllNuxtTasks = [];
-	let elements = await driver.$(taskSection);
-	await elements.$$(taskTitle).map(async function (element) {
-		listOfAllNuxtTasks.push(await element.getText());
-	});
-	return listOfAllNuxtTasks;
-}
-
 async function hoverOverTaskAndClickMenu(taskName) {
-	await driver.pause(5000);
 	let taskTitle = await driver.$(mod_extsprintf.sprintf(taskTitleText, taskName));
-	await taskTitle.scrollIntoView(false);
-	if (await taskTitle.isDisplayedInViewport()){
-		let xOffset = await taskTitle.getLocation('x');
-		let yOffset = await taskTitle.getLocation('y');
-		taskTitle.moveTo(xOffset, yOffset);
-		await driver.pause(5000);
-		await elementHelpers.click(mod_extsprintf.sprintf(taskActionMenu, taskName));
-		await driver.pause(3000);
+	await waitHelpers.waitUntilElementIsClickable(taskTitle);
+	if (!(taskTitle.isDisplayedInViewport())){
+		await elementHelpers.moveToElement(taskTitle);
 	}
+	await waitHelpers.waitUntilElementIsVisible(taskTitle);
+	await elementHelpers.moveToElement(taskTitle);;
+	await elementHelpers.click(mod_extsprintf.sprintf(taskActionMenu, taskName));
+
 }
 
 async function clickTaskOnActionMenu(button){
+	await waitHelpers.waitUntilElementIsClickable(getTaskActionBtnSelector(button));
 	await elementHelpers.hoverOverMenuOptions(getTaskActionBtnSelector(button))
 }
 
 async function taskTitleSelector(taskName){
```

Comment:
Instead of adding the method here and then making it public and using it in another file. Why not add it directly to the other file where it's needed and used? (TaskPage.js)

Reply:
If we use it directly in other file i.e. TaskPage.js. we have to repeat the selector again their. I made this function so we don't duplicate the selector in both files. The selector, I'm talking about is : `taskTitleText`. If you are okay with that I can remove it and will do it. Let me know about your decision.

Reply:
Thanks for your explanation.
Actually, I don't know what's best practice here. Duplicated selectors or make the method public... Now I'm actually wondering why if the method is correctly located in taskPage. Because taskPage is the page of a specific task and taskListPage is the page with all the tasks.
TaskPage.js was created to have eveything related to a task in there.